lib/objFile.ml

```diff
@@ -65,15 +65,18 @@ type segment = {
 type in_handle = {
   in_filename : string;
   in_channel : in_channel;
-  in_segments : segment CString.Map.t;
+  in_segments : Obj.t segment CString.Map.t;
 }
 
 type out_handle = {
   out_filename : string;
   out_channel : out_channel;
-  mutable out_segments : segment CString.Map.t;
+  mutable out_segments : Obj.t segment CString.Map.t;
 }
 
+type 'a id = { id : string }
+
+let make_id id = { id }
```

Comment:
Does it matter if this gets called twice with the same string?

Reply:
Don't care.